### PR TITLE
Show halves of checksums in correct order

### PR DESCRIPTION
**Pull request by vitlibar**
### Changelog category:
- Improvement

### Changelog entry:
Show halves of checksums in `system.parts`, `system.projection_parts` and in error messages in the correct order.

This PR changes only output representation of hashes. It changes the output representation into always using positional notation as we always do when we write numbers. For example the hash `ffeeddccbbaa99887766554433221100` will now always mean
```
(0xff << 120) +
(0xee << 112) +
(0xdd << 104) +
...
(0x22 << 16) +
(0x11 << 8) +
0x00
```

And before this PR it the same hash `ffeeddccbbaa99887766554433221100` could sometimes mean
```
(0x77 << 120) +
(0x66 << 112) +
(0x55 << 104) +
(0x44 << 96) +
(0x33 << 88) +
(0x22 << 80) +
(0x11 << 72) +
(0x00 << 64) +
(0xff << 56) +
(0xee << 48) +
(0xdd << 40) +
(0xcc << 32) +
(0xbb << 24) +
(0xaa << 16) +
(0x99 << 8) +
0x88
```

Also this PR introduces better names for member fields of the `CityHash_v1_0_2::uint128` struct (`first`, `second` were renamed to `low64`, `high64`).

### CODE REVIEW

File: base/base/wide_integer_impl.h [hunk 1]
```diff
@@ -281,6 +288,14 @@ struct integer<Bits, Signed>::_impl
         }
     }
 
+    constexpr static void wide_integer_from_cityhash_uint128(integer<Bits, Signed> & self, const CityHash_v1_0_2::uint128 & value) noexcept
```
Comment (vitlibar) [inline base/base/wide_integer_impl.h:291]: `CityHash_v1_0_2::uint128` is not tuple-like anymore so I had to add this function to implement conversion `CityHash_v1_0_2::uint128` -> `UInt128`.

**Review by antaljanosbenjamin — COMMENTED** [review]
In the changelog entry, could you please clarify that we use the BigEndian order (if I am not mistaken) instead of saying "correct order"?

Apart from this and the failing tests it looks good to me.

**Comment by vitlibar** [comment]
> In the changelog entry, could you please clarify that we use the BigEndian order (if I am not mistaken) instead of saying "correct order"?

It's like that, but not exactly. Big endian is about representation in memory, not in output. And this PR is only about output representation. It changes the output representation into always using positional notation as we always do when we write numbers.

I mean for example the hash `ffeeddccbbaa99887766554433221100` will now always mean
```
(0xff << 120) +
(0xee << 112) +
(0xdd << 104) +
...
(0x22 << 16) +
(0x11 << 8) +
0x00
```

And before this PR it the same hash `ffeeddccbbaa99887766554433221100` could sometimes mean
```
(0x77 << 120) +
(0x66 << 112) +
(0x55 << 104) +
(0x44 << 96) +
(0x33 << 88) +
(0x22 << 80) +
(0x11 << 72) +
(0x00 << 64) +
(0xff << 56) +
(0xee << 48) +
(0xdd << 40) +
(0xcc << 32) +
(0xbb << 24) +
(0xaa << 16) +
(0x99 << 8) +
0x88
```

**Review by antaljanosbenjamin — APPROVED** [review]
Thanks for the clarification, it makes sense!

**Comment by vitlibar** [comment]
Waiting
https://github.com/ClickHouse/ClickHouse/issues/50681
https://github.com/ClickHouse/ClickHouse/issues/51439
https://github.com/ClickHouse/ClickHouse/issues/51150

**Comment by vitlibar** [comment]
Test [test_ssl_cert_authentication/test.py::test_https_non_ssl_auth](https://s3.amazonaws.com/clickhouse-test-reports/51040/5570863676d158017a9df4d408575121a063bec8/integration_tests__asan__[6_6].html) is known to be flaky (https://github.com/ClickHouse/ClickHouse/issues/49400).

**Pull request merged**